app/src/main/java/com/example/designpatternstudy/Proxy/Protecting/ProxyRiderCommand.kt

```diff
@@ -0,0 +1,16 @@
+package com.example.designpatternstudy.Proxy.Protecting
+
+class ProxyRiderCommand : RiderCommand {
+    private var riderCommand: RiderCommandImpl = RiderCommandImpl()
```

Comment:
```suggestion
    private val riderCommand: RiderCommand by lazy { RiderCommandImpl() }
```
proxy의 service object는 지연 초기화를 하는게 적절해보입니다 :)